src/log.rs

```diff
@@ -22,6 +30,127 @@ impl Operation {
                 error,
                 latency: _,
             } => op.is_some() || error.is_some(),
+            Operation::CollectStats { per_dir_stats: _, loop_stats: _ } => {
+                true // logic punted to StatCollector
+            },
         }
     }
+
+    pub fn log_str(&mut self) -> String {
+        // This unwrap seems safe, afaict. We're not cramming any user supplied strings in here.
+        serde_json::to_string(self).unwrap()
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Stats {
+    dir_stats: Histo,
+    loop_stats: Histo,
+}
+
+/// A serializable form of a hdrhistogram, mainly just for logging out 
+/// in a way we want to read it
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Histo {
+    mean: f64,
+    count: u64,
+    min: u64,
+    max: u64,
+    percentiles: Vec<Percentile>,
 }
+
+/// For serializing to JSON
+///
+/// Choice of tiny names because this one shows up a lot, one
+/// for each percentile bucket. It shows a lot more data
+/// points at the upper percentiles, so we need to capture
+/// both percentile and associated millisecond value.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Percentile {
+    pct: f64,
+    val: u64,
+}
+
+impl Histo {
+    pub fn from_histogram(hist: &Histogram<u64>) -> Histo {
+        Self {
+            mean: hist.mean(),
+            count: hist.len(),
+            min: hist.min(),
+            max: hist.max(),
+            percentiles: hist.iter_quantiles(2).map(|q| Percentile {
+                pct: q.percentile(),
+                val: q.value_iterated_to(),
+            }).collect::<Vec<_>>(),
```

Comment:
I suspect since you're collecting this and assigning straight to `percentiles`, you can drop the turbofish on `collect` and it'll imply the type fine.

```suggestion
            }).collect(),
```